scheduler/integration-test/src/scheduler.test.js

```diff
@@ -1,52 +1,176 @@
-    expect(response.type).toEqual('application/json')
-    expect(response.body).toHaveLength(2)
-  })
+
+    const triggerRequests = await request(MOCK_SERVER_URL)
+      .get('/triggerRequests/1')
+
+    expect(triggerRequests.body).toBe(1)
+  }, TIMEOUT)
+
+  test('Should not trigger datasource after deletion event', async () => {
+    const channel = await createAmqpChannel()
+
+    const creationEvent = createDatasourceEvent(2, 2000, 1000, true)
+    const deletionEvent = createDeletionEvent(2)
+
+    channel.publish(AMQP_EXCHANGE, AMQP_DATASOURCE_CONFIG_CREATED_TOPIC, creationEvent)
+    channel.publish(AMQP_EXCHANGE, AMQP_DATASOURCE_CONFIG_DELETED_TOPIC, deletionEvent)
+
+    await sleep(3000)
+
+    const triggerRequests = await request(MOCK_SERVER_URL)
+      .get('/triggerRequests/2')
+
+    expect(triggerRequests.body).toBe(0)
+  }, TIMEOUT)
+
+  test('Should update trigger after update event', async () => {
+    const channel = await createAmqpChannel()
+
+    const creationEvent = createDatasourceEvent(3, 500, 10000, true)
+    const updateEvent = createDatasourceEvent(3, 500, 500, true)
+    console.log(updateEvent.toString())
+
+    channel.publish(AMQP_EXCHANGE, AMQP_DATASOURCE_CONFIG_CREATED_TOPIC, creationEvent)
+    await sleep(200)
+    channel.publish(AMQP_EXCHANGE, AMQP_DATASOURCE_CONFIG_UPDATED_TOPIC, updateEvent)
+
+    await sleep(4500)
+
+    const triggerRequests = await request(MOCK_SERVER_URL)
+      .get('/triggerRequests/3')
+
+    expect(triggerRequests.body).toBeGreaterThan(1)
+  }, TIMEOUT)
 })
 
+function createDeletionEvent (datasourceId) {
+  const event = {
+    datasource: {
+      id: datasourceId
+    }
+  }
+  return Buffer.from(JSON.stringify(event))
+}
+
+function createDatasourceEvent (datasourceId, delay, interval, periodic) {
+  const event = {
+    datasource: {
+      id: datasourceId,
+      trigger: {
+        firstExecution: jsonDateAfter(delay),
+        periodic,
+        interval
+      }
+    }
+  }
+  return Buffer.from(JSON.stringify(event))
+}
+
 function sleep (ms) {
   return new Promise(resolve => setTimeout(resolve, ms))
 }
 
-afterAll(() => setTimeout(() => process.exit(), 1000))
+const amqpConnect = async (amqpUrl, retries, backoff) => {
+  for (let i = 1; i <= retries; i++) {
+    console.info(`Connecting to AMQP broker (${i}/${retries})`)
+    try {
+      const connection = await AMQP.connect(amqpUrl)
+      console.log('Successfully established connection to AMQP broker.')
+      return connection
+    } catch (error) {
+      console.info(`Error connecting to AMQP broker: ${error}. Retrying in ${backoff} seconds`)
+      await sleep(backoff)
+    }
+  }
+  throw new Error('Could not establish connection to AMQP broker')
+}
+
+const createAmqpChannel = async () => {
+  const channel = await amqpConnection.createChannel()
+  await channel.assertExchange(AMQP_EXCHANGE, 'topic', { durable: true })
+  return channel
+}
+
+const logConfigs = () => {
```

Comment:
Should be done in env file

Reply:
Since the AMQP topics and the exchange name are defined in the test, this is not possible. 
Of course, we could also define them in the env vars, but I think this should be done in another pr. Especially since that would affect the other services' integration tests as well.

Reply:
I would also keep it in the `scheduler.test.js` file. This way it will only produce one console log output for the test. In previous solutions where the ENVs are logged from the `env.js` one by one it produced a lot of clutter.